### PR TITLE
USHIFT-1201: Transition to AWS

### PR DESCRIPTION
**Pull request by copejon**

### CODE REVIEW

File: ci-operator/config/openshift/microshift/openshift-microshift-main.yaml [hunk 1]
```diff
@@ -138,6 +138,24 @@ tests:
...
+- as: e2e-reboot-arm
+  cluster: build02
+  optional: true
+  run_if_changed: ^(assets/.*|cmd/.*|etcd/.*|pkg/.*|packaging/.*|vendor/.*|go\.sum|go\.mod|Makefile)
+  steps:
+    cluster_profile: aws
+    env:
+      EC2_AMI: ami-0c231be9fb222349e
+      EC2_INSTANCE_TYPE: a1.2xlarge
+    workflow: openshift-microshift-e2e-reboot-tests-aws
```
Comment (pacevedom) [inline ci-operator/config/openshift/microshift/openshift-microshift-main.yaml:158]: This is already part of the e2e test defined [here](https://github.com/openshift/release/blob/master/ci-operator/config/openshift/microshift/openshift-microshift-main.yaml#L141-L149).

File: ci-operator/config/openshift/microshift/openshift-microshift-main.yaml [hunk 2]
```diff
@@ -157,6 +175,25 @@ tests:
...
+- as: e2e-router-smoke-test-arm
+  cluster: build02
+  optional: true
+  run_if_changed: ^(assets/.*|cmd/.*|etcd/.*|pkg/.*|packaging/.*|vendor/.*|go\.sum|go\.mod|Makefile|.*\.sh)
+  steps:
+    cluster_profile: aws
+    env:
+      EC2_AMI: ami-0c231be9fb222349e
+      EC2_INSTANCE_TYPE: a1.2xlarge
+    workflow: openshift-microshift-e2e-router-smoke-test-aws
```
Comment (pacevedom) [inline ci-operator/config/openshift/microshift/openshift-microshift-main.yaml:196]: And same goes here.

File: ci-operator/config/openshift/microshift/openshift-microshift-release-4.13.yaml [hunk 3]
```diff
@@ -115,76 +115,95 @@ tests:
...
+- as: microshift-e2e-arm
+  cluster: build02
+  optional: true
+  run_if_changed: ^(assets/.*|cmd/.*|etcd/.*|pkg/.*|packaging/.*|vendor/.*|go\.sum|go\.mod|Makefile|.*\.sh)
+  steps:
+    cluster_profile: aws
+    env:
+      EC2_AMI: ami-0c231be9fb222349e
+      EC2_INSTANCE_TYPE: a1.2xlarge
+    workflow: openshift-microshift-e2e-microshift-tests-aws
```
Comment (pacevedom) [inline ci-operator/config/openshift/microshift/openshift-microshift-release-4.13.yaml:187]: These should only be part of 4.14+ versions, as they are not defined in release-4.13 or lower branches in microshift.

File: ci-operator/config/openshift/microshift/openshift-microshift-release-4.14.yaml [hunk 4]
```diff
@@ -139,6 +139,24 @@ tests:
...
+- as: e2e-reboot-arm
+  cluster: build02
+  optional: true
+  run_if_changed: ^(assets/.*|cmd/.*|etcd/.*|pkg/.*|packaging/.*|vendor/.*|go\.sum|go\.mod|Makefile)
+  steps:
+    cluster_profile: aws
+    env:
+      EC2_AMI: ami-0c231be9fb222349e
+      EC2_INSTANCE_TYPE: a1.2xlarge
+    workflow: openshift-microshift-e2e-reboot-tests-aws
```
Comment (pacevedom) [inline ci-operator/config/openshift/microshift/openshift-microshift-release-4.14.yaml:159]: Same as main branch comment.

File: ci-operator/config/openshift/microshift/openshift-microshift-release-4.14.yaml [hunk 5]
```diff
@@ -158,6 +176,25 @@ tests:
...
+- as: e2e-router-smoke-test-arm
+  cluster: build02
+  optional: true
+  run_if_changed: ^(assets/.*|cmd/.*|etcd/.*|pkg/.*|packaging/.*|vendor/.*|go\.sum|go\.mod|Makefile|.*\.sh)
+  steps:
+    cluster_profile: aws
+    env:
+      EC2_AMI: ami-0c231be9fb222349e
+      EC2_INSTANCE_TYPE: a1.2xlarge
+    workflow: openshift-microshift-e2e-router-smoke-test-aws
```
Comment (pacevedom) [inline ci-operator/config/openshift/microshift/openshift-microshift-release-4.14.yaml:197]: Same as main branch comment.

File: ci-operator/config/openshift/microshift/openshift-microshift-release-4.15.yaml [hunk 6]
```diff
@@ -138,6 +138,24 @@ tests:
...
+- as: e2e-reboot-arm
+  cluster: build02
+  optional: true
+  run_if_changed: ^(assets/.*|cmd/.*|etcd/.*|pkg/.*|packaging/.*|vendor/.*|go\.sum|go\.mod|Makefile)
+  steps:
+    cluster_profile: aws
+    env:
+      EC2_AMI: ami-0c231be9fb222349e
+      EC2_INSTANCE_TYPE: a1.2xlarge
+    workflow: openshift-microshift-e2e-reboot-tests-aws
```
Comment (pacevedom) [inline ci-operator/config/openshift/microshift/openshift-microshift-release-4.15.yaml:158]: Same as main branch comment.

File: ci-operator/config/openshift/microshift/openshift-microshift-release-4.15.yaml [hunk 7]
```diff
@@ -157,6 +175,25 @@ tests:
...
+- as: e2e-router-smoke-test-arm
+  cluster: build02
+  optional: true
+  run_if_changed: ^(assets/.*|cmd/.*|etcd/.*|pkg/.*|packaging/.*|vendor/.*|go\.sum|go\.mod|Makefile|.*\.sh)
+  steps:
+    cluster_profile: aws
+    env:
+      EC2_AMI: ami-0c231be9fb222349e
+      EC2_INSTANCE_TYPE: a1.2xlarge
+    workflow: openshift-microshift-e2e-router-smoke-test-aws
```
Comment (pacevedom) [inline ci-operator/config/openshift/microshift/openshift-microshift-release-4.15.yaml:196]: Same as main branch comment.

**Review by pacevedom — COMMENTED** [review]
Looking good overall. Maybe we could merge the -aws with non-aws workflows by doing a platform discovery in order to get the right commands to retrieve the IP, but that can be done later.

**Comment by copejon** [comment]
Current failures are a result of the service-ca container fail to come up due to what looks like selinux permissions:
```
> $ oc logs -n openshift-service-ca pod/service-ca-7754867754-c27hs service-ca-controller
exec container process `/usr/bin/service-ca-operator`: Operation not permitted 
```

**Comment by copejon** [comment]
These are expected to fail until https://github.com/openshift/microshift/pull/1742 is merged

File: ci-operator/config/openshift/microshift/openshift-microshift-release-4.13.yaml [hunk 8]
```diff
@@ -97,9 +97,18 @@ tag_specification:
 tests:
 - as: verify
   cluster: build02
-  commands: make verify
-  container:
-    from: src
+  steps:
+    test:
+    - as: verify
+      commands: make verify-ci
```
Comment (pacevedom) [inline ci-operator/config/openshift/microshift/openshift-microshift-release-4.13.yaml:103]: Keep in mind that new static analysis only runs on 4.14+. This target does not exist in this branch.

File: ci-operator/config/openshift/microshift/openshift-microshift-release-4.13.yaml [hunk 9]
```diff
@@ -115,115 +124,43 @@ tests:
...
+    cluster_profile: aws
     env:
-      GOOGLE_COMPUTE_IMAGE_NAME: rhel-9-arm64-v20230306
-      GOOGLE_COMPUTE_MACHINE_TYPE: t2a-standard-4
-    workflow: openshift-microshift-e2e-greenboot
-- as: e2e-router-smoke-test
+      EC2_AMI: ami-0c231be9fb222349e
+      EC2_INSTANCE_TYPE: a1.2xlarge
+    workflow: openshift-microshift-e2e-openshift-conformance-reduced-aws
+- as: microshift-e2e
```
Comment (pacevedom) [inline ci-operator/config/openshift/microshift/openshift-microshift-release-4.13.yaml:141]: This test only exists from 4.14. You need to keep the reboot, greenboot, load-balancer and router smoke tests in this branch.

File: ci-operator/config/openshift/microshift/openshift-microshift-release-4.13.yaml [hunk 10]
```diff
@@ -115,115 +124,43 @@ tests:
...
-  cluster: build02
-  optional: true
-  run_if_changed: ^(assets/.*|cmd/.*|etcd/.*|pkg/.*|packaging/.*|vendor/.*|go\.sum|go\.mod|Makefile|.*\.sh)
-  steps:
-    cluster_profile: gcp
-    env:
-      GOOGLE_COMPUTE_IMAGE_NAME: rhel-9-arm64-v20230306
-      GOOGLE_COMPUTE_MACHINE_TYPE: t2a-standard-4
-    workflow: openshift-microshift-e2e-loadbalancer-smoke-test
-- as: nightly-conformance-parallel
```
Comment (pacevedom) [inline ci-operator/config/openshift/microshift/openshift-microshift-release-4.13.yaml:211]: Need to keep these nightlies for reporting to testgrid.

**Comment by copejon** [comment]
These tests are running the openshift/conformance/[parallel/serial] test suites, which depend on CRDs from the openshift cluster-config-operator.  So these tests will never pass.

ci/rehearse/periodic-ci-openshift-microshift-release-4.13-nightly-conformance-parallel
ci/rehearse/periodic-ci-openshift-microshift-release-4.13-nightly-conformance-serial

**Comment by copejon** [comment]
/pj-rehearse pull-ci-openshift-microshift-release-4.13-e2e-openshift-conformance-reduced-arm

**Comment by pacevedom** [comment]
/pj-rehearse pull-ci-openshift-microshift-release-4.13-e2e-loadbalancer-smoke-test-arm

**Comment by pacevedom** [comment]
/pj-rehearse periodic-ci-openshift-microshift-release-4.13-nightly-conformance-serial

**Review by pacevedom — COMMENTED** [review]
/lgtm

**Pull request merged**